### PR TITLE
Fix build on iOS

### PR DESCRIPTION
**Pull request by nakajima**
We were using an outdated version of the RN push notification library for iOS. This PR updates to use the most recent version.

### CODE REVIEW

File: example/ios/Podfile [hunk 1]
```diff
@@ -49,6 +49,7 @@ target 'xmtpreactnativesdkexample' do
   flags = get_default_flags()
 
   pod 'secp256k1.swift', :modular_headers => true
+  pod 'RNCPushNotificationIOS', :path => '../node_modules/@react-native-community/push-notification-ios'
```
Comment (michaelx11) [inline example/ios/Podfile]: this looks brittle - the pod doesn't get pulled in normally? i.e. via expo hooks checking npm packages

**Review by michaelx11 — APPROVED** [review]
LGTM - one question around the Podfile path-based item

File: example/ios/Podfile (hunk 1 above)
Comment (nakajima) [reply example/ios/Podfile]: It was something I pulled in from the docs here: https://reactnative.dev/docs/pushnotificationios.html, after removing it stuff still worked though so I don't think we need it. Removed in https://github.com/xmtp/xmtp-react-native/pull/31/commits/c97c0e93175ae126833d67cdb88eb467106a9900.

Comment (nakajima) [reply example/ios/Podfile]: Then actually removed in https://github.com/xmtp/xmtp-react-native/pull/31/commits/655762c2408345f07cc98364b21f4972fd49d309.

**Review by nplasterer — APPROVED** [review]
Sorry about breaking that!

**Review by daria-github — APPROVED** [review]
Amazing that this did it, thanks for finding the fix.

**Pull request merged**